pkg/apply/taskrunner/runner.go

```diff
@@ -236,7 +243,15 @@ func (b *baseRunner) run(ctx context.Context, taskQueue chan Task,
 			}
 			if msg.Err != nil {
 				b.amendTimeoutError(taskContext, msg.Err)
-				return msg.Err
+				// If `o.continueOnError` is `false`, we return `msg.Err` immediately, and the caller of the function
+				// would send a ErrorType Event including `msg.Err` to eventChannel.
+				// If `o.continueOnError` is `true`, we don't not return immediately, but
+				// instead log the error and send a WaitType Event including `msg.Err` to eventChannel before continuing.
+				if !o.continueOnError {
+					return msg.Err
+				}
+				klog.Infof("ignore the TimeoutError: %v", msg.Err)
```

Comment:
This log message is inaccurate. Errors from the task channel are not always wait task timeouts. Suggest: `continuing on error (task: %q): %v` with the `currentTask.Name()`.

Reply:
Good catch. Done.